File: boj/1002/soln1002.go
/*
// 백준
// No. 1002
// Go 1.24.8
// by "nno0obb"
*/

package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func logic(x1, y1, r1, x2, y2, r2 int) int {
	d := math.Sqrt(math.Pow(float64(x1-x2), 2) + math.Pow(float64(y1-y2), 2))
	if d == 0 {
		if r1 == r2 {
			return -1
		} else {
			return 0
		}
	} else {
		if float64(r1+r2) < d {
			return 0
		} else if float64(r1+r2) == d {
			return 1
		} else if math.Abs(float64(r1-r2)) < d && d < float64(r1+r2) {
			return 2
		} else if math.Abs(float64(r1-r2)) == d {
			return 1
		} else {
			return 0
		}
	}
}

func main() {
	defer writer.Flush()

	// Input
	var T int
	fmt.Fscanln(reader, &T)
	for range T {
		var x1, y1, r1, x2, y2, r2 int
		fmt.Fscanln(reader, &x1, &y1, &r1, &x2, &y2, &r2)

		// Logic
		ans := logic(x1, y1, r1, x2, y2, r2)

		// Output
		fmt.Fprintln(writer, ans)
	}
}

File: problem_solving/boj/1002/soln1002.py
"""
# 백준
# No. 1002 
# Python 3.11.9
# by "nno0obb"
"""

from math import sqrt


def logic(x1, y1, r1, x2, y2, r2):

    d = sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)

    if d == 0:
        if r1 == r2:
            return -1
        elif r1 != r2:
            return 0

    if d > 0:
        if r1 + r2 < d:
            return 0
        elif r1 + r2 == d:
            return 1
        elif abs(r1 - r2) < d < r1 + r2:
            return 2
        elif abs(r1 - r2) == d:
            return 1
        elif abs(r1 - r2) > d:
            return 0


def main():
    # Input
    T = int(input())
    for _ in range(T):
        x1, y1, r1, x2, y2, r2 = map(int, input().split())

        # Logic
        ans = logic(x1, y1, r1, x2, y2, r2)

        # Output
        print(ans)


if __name__ == "__main__":
    main()
